exe/Wrapper.hs

```diff
@@ -142,3 +163,35 @@ findProjectCradle = do
     Nothing -> hPutStrLn stderr "No 'hie.yaml' found. Try to discover the project type!"
 
   Session.loadCradle def hieYaml d
+
+data WrapperSetupError
+    = FailedToObtainGhcVersion (ActionName Void) CradleError
+    | NoneCradleGhcVersion (ActionName Void)
+    | NoLanguageServer String [FilePath]
+    | ToolRequirementMissing String (ActionName Void)
+    deriving (Show)
+
+data Shorten = Shorten | NoShorten
+
+-- | Pretty error message displayable to the future.
+-- Extra argument 'Shorten' can be used to shorten error message.
+-- Reduces usefulness, but allows us to show the error message via LSP
+-- as LSP doesn't allow any newlines and makes it really hard to read
+-- the message otherwise.
+prettyError :: WrapperSetupError -> Shorten ->  T.Text
+prettyError (FailedToObtainGhcVersion name crdlError) shorten =
+  "Failed to find the GHC version of this " <> T.pack (show name) <> " project." <>
+  case shorten of
+    Shorten ->
+      "\n" <> T.pack (fromMaybe "" . listToMaybe $ cradleErrorStderr crdlError)
+    NoShorten ->
+      "\n" <> T.pack (intercalate "\n" (cradleErrorStderr crdlError))
+prettyError (NoneCradleGhcVersion name) _ =
+  "Failed to get the GHC version of the " <> T.pack (show name) <>
+  " project, since we have a none cradle"
+prettyError (NoLanguageServer ghcVersion candidates) _ =
+  "Failed to find a HLS version for GHC " <> T.pack ghcVersion <>
+  "\nExecutable names we failed to find: " <> T.pack (intercalate "," candidates)
```

Comment:
This wording feels bad

Reply:
`FailedToObtainGhcVersion` and `NoneCradleGhcVersion` look they could be identical? i.e. something like `Failed to find GHC version for project: {name}` (unless name implies something else?) and then for `NoneCradle` keep the same but with `(NoneCradle)` or some other marker attached?

Reply:
They are slightly different though, we should improve the error message!
* In the case of a `none` cradle, we can not figure out the GHC version to use, thus can't decide which HLS version we want to launch. The outside world specifically instructed us not to load this project, probably via a `hie.yaml`
* FailedToObtainGhcVersion entails that some command has failed invocation, e.g. `cabal v2-exec -- ghc -v0 --numeric-version`, e.g. we attempted to load the project, but we couldn't.

Reply:
Ah I was paying close enough attention I didn't see the extra details added to